birdyboard.py

```diff
@@ -0,0 +1,421 @@
+                    self.user_name = self.usurper.user_library[self.user_id]["user_name"]
+                    print("logging in " + self.usurper.user_library[self.user_id]["real_name"] + "\n Welcome " + self.user_name + "!")
+                    print(self.logged_in_options)
+                    self.logged_in_menu()
+                except KeyError:
+                    print("your input is not in the list of users.")
+                    self.users_menu()
+                finally:
+                    pass
+
+# #####################################
+# ######## VIEW THREADS ###############
+# #####################################
+
+    def view_threads_menu(self):
+        """
+        menu that appears after all chirp threads are printed (threader.generate_threads_list). Allows the user to view a full chirp thread based on the index, or to go back or exit.
+        arguments: none
+        """
+
+        # deal with the fact that un-logged in users can see public chirp threadsbut only logged-in users can add a new thread. Also handle what happens if there are no threads to view.
+        logged_in = False
+        if len(self.user_name) > 0:
+            logged_in = True
+
+        print("'n' for new chirp thread.\n'b' to go back.\n'x' to exit.")
+
+        if len(self.threader.temp_threads) == 0:
+            print("no private chirps yet!")
+        else:
+            print("enter the number of the chirp you'd like to view the full thread for.")
+        if logged_in is True:
+            thread = input(">> ")
+        else:
+            thread = input(">> ")
+
+        # handle input.
+        if thread == "b":  # go back to either logged in or unlogged in menu.
+            print("going back.")
+            if logged_in is True:
+                print(self.logged_in_options)
+                self.logged_in_menu()
+            else:
+                print(self.unlogged_in_options)
+                self.unlogged_in_menu()
+        elif thread == "x":  # exit.
+            print("goodbye.")
+            exit()
+        elif thread == "n":  # allow a user to create a new thread, but not an unlogged-in user.
+            if logged_in is True:
+                if self.public_or_private == "public":  # create a new public or private thread.
+                    self.new_public_thread_menu()
+                else:
+                    self.usurper.generate_users_list()
+                    self.new_private_thread_menu()
+            else:
+                pass
+        else:  # select the number of the printed tweet.
+            try:
+                thread = int(thread)
+            except ValueError:
+                print("you didn't enter a number.")
+                self.view_threads_menu_thread()
+            finally:
+                try:
+                    self.thread_id = self.threader.temp_threads[thread]
+                    self.chirper.generate_chirp_list(self.thread_id)
```

Comment:
Doesn't need testing just because it has an input.  Both this and the user_list one are essentially menu extension methods, they don't really need testing.